### PR TITLE
linting: fixed some linting issues

### PR DESCRIPTION
**Pull request by fredbi**
Signed-off-by: Frederic BIDON <fredbi@yahoo.com>

### CODE REVIEW

File: analyzer.go [hunk 1]
```diff
@@ -237,7 +237,8 @@ func (s *Spec) initialize() {
 			s.authSchemes[k] = struct{}{}
 		}
 	}
-	for path, pathItem := range s.AllPaths() {
+	for path, v := range s.AllPaths() {
+		pathItem := v
```
Comment (gregmarr) [inline analyzer.go]: Why this change?

Comment (fredbi) [reply analyzer.go]: updated linter detects a code smell when using pointers on ranged value (https://github.com/kyoh86/scopelint). This keeps it happy, even though in all the "fixed" cases the bug-prone behavior did not show up (I assume that values of type slice and maps are not subject to this).

I preferred to apply this rather than disabling the scopelint check altogether, since this might prove useful. What do you think @gregmarr ?

Comment (gregmarr) [reply analyzer.go]: Seems to be trading one smell for another.  Also it seems like it has the potential to introduce a lot of extra work at runtime.  If it's a large value, then we're copying it unnecessarily.  If it's already a pointer/ref, then this change doesn't seem like it would provide any additional safety.  Without a comment here that it's being done to keep a linter happy, then it's prime fodder for someone to clean up later to remove the unneeded local variable.  Unfortunately, this is one of those categories where you can get a lot of false positives without deeper inspection of the called function, but the real positives are very useful.

Comment (fredbi) [reply analyzer.go]: It is not a pointer, but a reference. So the copy (here a map) remains cheap. When you range pointers, you usually don't use the pointer's pointer in the loop, so this is not happening.

It seems you were careful enough not to stumble upon bugs cause by this. I did once, before knowing of this "pin the range variable", and you really can search for quite some time before you find out (without being prior informed about this peculiar inner workings of the range operator).

The idea with this linter is that it wants to impose the "variable pinning" as idiomatic when you get to use a pointer in the loop.

I am okay with the idea to put more comment on this.

Comment (fredbi) [reply analyzer.go]: ok @gregmarr , after some more digging and testing I now realize that this is definitely a false positive from the linter.
I am reverting these unwise changes.

Thank you for taking your time challenging this!

Comment (gregmarr) [reply analyzer.go]: I had never heard of the "variable pinning", but I can see where it can be useful.  I think it's niche enough that it would need local documentation on why it's there.  Coming from the C & C++ side, I'm always very careful about what is done with local variables, whether they're obvious local variables or loop variables.  This doesn't seem to me to be any safer than what was there before, but maybe it is.

Comment (fredbi) [reply analyzer.go]: But this is real. However, in our case, the code is safe and the linter reports a false positive.
Look https://gist.github.com/fredbi/423db4ba3cef139e42c66f643aa8d661
It is really common to stumble into this in test programs.

Comment (gregmarr) [reply analyzer.go]: My question is whether or not this bug:
	for k, v := range a {
		result = append(result, &v)
	}
is actually fixed by changing it to this:
	for k, v := range a {
		safev := v
		result = append(result, &safev)
	}

Comment (gregmarr) [reply analyzer.go]: Note that any further discussion here is just for my own education, so thanks for the discussion. :)

Comment (fredbi) [reply analyzer.go]: @gregmarr Yes this is the commonly adopted idiom to address the "single pointer" issue. You can verify by yourself with the small gist I shared above.

This technique is dubbed "variable pinning" and this what the linter (now disabled with this PR) attempts to enforce, unfortunately with some false positive that cannot be sorted out manually.
